tests/lib/rules/no-typos.js

```diff
@@ -763,5 +754,17 @@ ruleTester.run('no-typos', rule, {
     }, {
       message: 'Typo in declared prop type: objectof'
     }]
+  }, {
+  // PropTypes declared on a component that is detected through JSDoc comments and is
+  // declared AFTER the PropTypes only works when ESLint version is >= 4.7.0
```

Comment:
We should support both eslint < v4.7, and >= v4.7, and this code should be made to fail as expected in both ranges.

Reply:
Hmm - not sure how we can make it fail with ESLint >= 4.7 😄 

It does feel a bit strange to really force a working case to be invalid though? Currently we do support both version ranges, but the plugin behaves differently in both... but yes, I agree that's kind of confusing as well.

I'll see what can be done.

Reply:
To clarify; I think that the code warning or not warning should behave identically in 4.6 and 4.7 - I don't really care which we pick, as long as it doesn't change.

If that's not possible, then that will mean 4.7 was a breaking change, and eslint core needs to revert something so that it again works as expected.

Reply:
We definitely cannot easily detect this case with ESLint 4.6. That's why we had to choose to ignore it until the ESLint issue fixed.

Indeed, 4.7 did some kind of internal refactor of ESLint core. If you're interested, this is the change: https://github.com/eslint/eslint/pull/9283

Reply:
in that case, we need to explicitly ignore it in eslint `^4.7`.